packages/react-admin/src/form/Autocomplete.tsx

```diff
@@ -0,0 +1,68 @@
+import { TextField } from "@material-ui/core";
+import MuiAutocomplete, { AutocompleteProps } from "@material-ui/lab/Autocomplete";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+
+interface IProps extends FieldRenderProps<string, HTMLInputElement | HTMLTextAreaElement> {
+    optionValue?: string;
+    optionLabel?: string;
+}
+
+interface AutocompleteAsyncProps<T> {
+    open: boolean;
+    options: T[];
+    loading: boolean;
+    onOpen: (event: React.ChangeEvent<{}>) => void;
+    onClose: (event: React.ChangeEvent<{}>) => void;
+}
+
+export function useAutocompleteAsyncProps<T>(loadOptions: () => Promise<T[]>): AutocompleteAsyncProps<T> {
+    const [open, setOpen] = React.useState(false);
```

Comment:
why do we need our own open state? isn't that duplicating Autocomplete's state without lifting it up?

Reply:
The open state is being lifted up through the hook and ist necessary to set the loading state properly (which is only shown when open=true and options=[]). The Hook is taking care of this.
Nevertheless, this code is originally taken from https://material-ui.com/components/autocomplete/#asynchronous-requests